### DISCUSSION TITLE
Automation - Only works once after 'test'

### DISCUSSION THREAD

**Discussion by tokkisva** [Budibase/budibase#8592] [Help] [answered]
Hi all,

Using self-hosted, bb version 2.1.9. This is the only automation I have, so no other traffic is occurring. 

I have an odd issue where my automation only works correctly one time after running 'Run Test'. 

Goal of this automation - generate multiple API POSTs to external data connector from row values gathered from form. The first API POST occurs on action (button click) when user submits form data to row. The second API POST should occur 1 second after a row is created.
The initial API POST always runs successfully, the second one performed by Automation only works one time after 'Run Test' is performed.

More or less, any subsequent rows created after a fresh 'Run Test' and form submission do not trigger the automation. Am I missing something?
My automation log is also empty despite having successful runs.

**Here's the order of events:**
User performs 'Run Test' on automation (success).
User creates submission from form, automation works correctly on first submission. (Both API POSTs run, one from the button on form and one from the automation)
User creates another submission from form, automation does not run at all. 

If I go back to the automation, and do 'Run Test', and then go back to the form and fill it out, it works! But just that one time.

**Automation Steps:**

1 When this happens: 
ROW CREATED

2 Do this:
Delay - 1000ms

3 Do this: 
External Data Connector -> REST API POST using values from submission row.

4 Do this: 
Backend Log

Automation Screenshot

![image](https://user-images.githubusercontent.com/38924262/200585153-749deb71-8eba-43ae-8ca2-562153bb2730.png)

Thoughts?

Thanks in advance!

**Comment by melohagan** [comment, marked as answer]
Hey @tokkisva 

That's unusual. One thing I would say, which probably isn't related to your issue, but the delay shouldn't make any difference to your binding values FYI.
So the `trigger.row` will have the value of at the point of the row being created, even if that row was altered within those 1000ms.

Just wondering, how much time has passed between the two submissions?

**Reply by tokkisva** [reply]
Hi Melohagan, 

It's soo odd! Even if I wait one day between submissions, the automation never kicks off even after hours of waiting. Unless I perform 'Run Test' again- and then fill out another submission, automation never runs. Do I have to publish the form first for the automation to run? I might move my dev instance to a dedicated server and reconfigure to see if I have the same issue.

Noted on the delay, it is there I don't get hit with rate limiting against the app I am POSTing to.

**Reply by melohagan** [reply]
Well your app will need to be published for automations to run. Automations don't run for unpublished apps (with the exception of Run Test)

**Reply by tokkisva** [reply]
This is embarrassing.. that did it, all I had to do was publish it and all is well. Thank you so much melohagan!!!!!!

**Reply by melohagan** [reply]
That's great! Not embarrassing at all. I realised the documentation didn't really specify that clearly, so I've added a callout to the introduction page here:
https://docs.budibase.com/docs/automation-steps
